packages/client/vite.config.ts

```diff
@@ -70,51 +70,48 @@ function targetBuildPlugin(dev: boolean, target: string): Plugin {
         const mockPath = `${file.dir}/${file.name}.${target}.${extension}`
         return this.resolve(mockPath, importer)
       } else {
-        const rootPrefix = "src/client/"
-        const thisImporter = (importer || "").replace(/\\/g, "/")
         if (
           !importer ||
-          !thisImporter.includes(rootPrefix) ||
+          importer.includes("node_modules") ||
           source === "./main"
         ) {
           return null
         }
 
-        const importedFile = path.parse(source)
+        if (!source.startsWith(".") && !source.startsWith("/")) {
+          return null
+        }
+
+        const sourcePath = path.parse(source)
+        const importerPath = path.parse(importer.replace(/\\/g, "/"))
 
-        const importerFile = path.parse(thisImporter)
+        // If imported file starts with /src we can not append it to importer dir
+        // so we need to strip the path by the rootPrefix first
+        const aliasedSourceRootPrefix = "/src"
+        const baseCandidatePath =
+          sourcePath.dir.startsWith(aliasedSourceRootPrefix) &&
+          importerPath.dir.includes(aliasedSourceRootPrefix)
+            ? `${importerPath.dir.split(aliasedSourceRootPrefix)[0]}/`
+            : importerPath.dir
+        const targetFileName = `${sourcePath.name}.${target.toLowerCase()}`
 
         const candidatePath = path.join(
-          // If imported file starts with /src we can not append it to importer dir
-          // so we need to strip the path by the rootPrefix first
-          importedFile.dir.startsWith("/src") &&
-            importerFile.dir.includes(rootPrefix)
-            ? `${importerFile.dir.split(rootPrefix)[0]}/`
-            : importerFile.dir,
-
-          importedFile.dir,
-          `${importedFile.name}.${target.toLowerCase()}`,
+          baseCandidatePath,
+          sourcePath.dir,
+          targetFileName,
         )
 
         // Source doesn't have file extension, so try all extensions
-        let candidate: string | null = null
-        const extensions = ["ts", "tsx"]
-        for (let i = 0; i < extensions.length; i++) {
-          try {
-            candidate = `${candidatePath}.${extensions[i]}`
-            statSync(candidate)
-            console.log("candidate good", candidate)
-          } catch (e) {
-            // console.log("Error with candidate", candidate, e)
-            candidate = null
-          }
-
-          if (candidate) {
-            break
-          }
+        const candidateTs = `${candidatePath}.ts`
```

Comment:
for ref, down to 18 candidate imports, with 7 distinct replacements (for openfin target)
if we ever amend this logic, we need to ensure that we don't cut any out